From f81753f09997e25e3d7c87a10fbea5e8662fed3c Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Sun, 22 May 2016 08:53:29 +0300
Subject: [PATCH 22/46] PCIe: imx6: refine link up error handle

When PCIe link does not come up do not disabe PCIe bus clock unless
both EP and RC modes are disabled.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 drivers/pci/host/pci-imx6.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/host/pci-imx6.c b/drivers/pci/host/pci-imx6.c
index c4e3b81..fce28ad 100644
--- a/drivers/pci/host/pci-imx6.c
+++ b/drivers/pci/host/pci-imx6.c
@@ -517,7 +517,9 @@ static int imx6_pcie_wait_for_link(struct pcie_port *pp)
 
 		if (!IS_ENABLED(CONFIG_PCI_IMX6_COMPLIANCE_TEST)) {
 			clk_disable_unprepare(imx6_pcie->pcie);
-			clk_disable_unprepare(imx6_pcie->pcie_bus);
+			if (!IS_ENABLED(CONFIG_EP_MODE_IN_EP_RC_SYS) &&
+			    !IS_ENABLED(CONFIG_RC_MODE_IN_EP_RC_SYS))
+				clk_disable_unprepare(imx6_pcie->pcie_bus);
 			clk_disable_unprepare(imx6_pcie->pcie_phy);
 			if (is_imx6sx_pcie(imx6_pcie))
 				clk_disable_unprepare(imx6_pcie->pcie_inbound_axi);
-- 
1.9.1

